### PR TITLE
Migrate `run-make/doctests-keep-binaries` to new rmake.rs format

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of https://github.com/rust-lang/rust/issues/121876.

r? @jieyouxu

### CODE REVIEW

File: src/tools/run-make-support/src/rustdoc.rs [hunk 1]
```diff
@@ -62,6 +62,12 @@ impl Rustdoc {
         self
     }
 
+    /// Set the path where the command will be run.
+    pub fn current_dir<P: AsRef<Path>>(&mut self, path: P) -> &mut Self {
```
Comment (GuillaumeGomez) [inline src/tools/run-make-support/src/rustdoc.rs:66]: Should I add this method in `Rustc` too?

Comment (jieyouxu) [reply src/tools/run-make-support/src/rustdoc.rs:66]: That makes sense, or even in the `impl_helper_methods` macro to generate that for all command wrappers.

File: tests/run-make/doctests-keep-binaries/rmake.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+fn main() {
+    setup_test_env(|out_dir, extern_path| {
+        rustdoc()
+            .input("t.rs")
+            .arg("-Zunstable-options")
+            .arg("--test")
+            .arg("--persist-doctests")
+            .arg(out_dir)
+            .arg("--extern")
+            .arg(format!("t={}", extern_path.display()))
```
Comment (jieyouxu) [inline tests/run-make/doctests-keep-binaries/rmake.rs:33]: Do we want any helper methods on `Rustdoc` for some of the stable flags? e.g.

```rs
rustdoc()
    // ...
    .extern("t", extern_path) // or whatever
```

**Review by jieyouxu — COMMENTED** [review]
Looks good, just some nitpicks.

File: tests/run-make/doctests-keep-binaries/rmake.rs (hunk 2 above)
Comment (GuillaumeGomez) [reply tests/run-make/doctests-keep-binaries/rmake.rs:33]: Hum... Considering there are only 3 doctests in the `run-make` testsuite, is it really worth it?

**Comment by GuillaumeGomez** [comment]
Moved the implementation of `current_dir` into the `impl_common_helpers` macro.

**Review by jieyouxu — APPROVED** [review]
Thanks! I like the `setup_test_env` helper :3

Feel free to r=me after CI is green.

**Comment by matthiaskrgr** [comment]
@bors r-
may need reformatting  https://github.com/rust-lang/rust/pull/124673#issuecomment-2093549140

**Comment by GuillaumeGomez** [comment]
Now we can put it into the rollup:

@bors r=jieyouxu rollup

**Pull request merged**